### PR TITLE
Smarter positioning of Spatial Debugger tag elements

### PR DESCRIPTION
**Pull request by Vix2021**
#### Description
Centre the Spatial Debugger tags over the actor's world location no matter how many parts of the tag (ex. Authority, Lock, etc) are enabled.

For the actors in the world:
1. Calculate the total width of the icons and text to be rendered
2. Calculate the horizontal offset based on the total width
3. Draw icons and text based on the calculated offset

#### Release note
I think this is too minor for a release note.

#### Tests
Tested manually with a local deployment of the Handover Gym by viewing different combinations of tags (ex. Authority, Lock, etc). 
See screenshots below:

solution_final
solution_final_a
solution_final_b
solution_final_c
solution_final_d

QA Testing
As above (Press F9 to activate the Spatial Debugger runtime menu)

#### Documentation
Updated the Spatial Debugger animation in https://docs.google.com/document/d/1G9cRz891Ycaiax-M7Psx3rlfX8qTIi0CTk8_71PL4Zg/edit#heading=h.j4ppr52lw6ck

### CODE REVIEW

**Review by alastairdglennie — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp [hunk 1]
```diff
@@ -460,13 +456,52 @@ void ASpatialDebugger::DrawTag(UCanvas* Canvas, const FVector2D& ScreenLocation,
...
 	const SpatialDebugging* DebuggingInfo = NetDriver->StaticComponentView->GetComponentData<SpatialDebugging>(EntityId);
 
-	static const float BaseHorizontalOffset(16.0f);
-
 	if (!FApp::CanEverRender()) // DrawIcon can attempt to use the underlying texture resource even when using nullrhi
 	{
 		return;
 	}
 
+	static const float BaseHorizontalOffset(16.0f);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:464]:
```suggestion
	static const float BaseHorizontalOffset = 16.0f;
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp [hunk 2]
```diff
@@ -460,13 +457,52 @@ void ASpatialDebugger::DrawTag(UCanvas* Canvas, const FVector2D& ScreenLocation,
...
+	{
+		// If tag should be centered, calculate the total width of the icons and text to be rendered
+		float TagWidth = 0;
+		if (bShowLock)
+		{
+			TagWidth += BaseHorizontalOffset;
+		}
+		if (bShowAuth)
+		{
+			TagWidth += BaseHorizontalOffset;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:483]: Why do we double increment here? Shouldn't AuthWidth just be +=1.f ?

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:483]: Oh it's used below

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:483]: Still a little confused - is this additional spacing for something?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp [hunk 3]
```diff
@@ -606,7 +638,7 @@ void ASpatialDebugger::DrawDebug(UCanvas* Canvas, APlayerController* /* Controll
 					continue;
 				}
 
-				DrawTag(Canvas, ScreenLocation, EntityId, Actor->GetName());
+				DrawTag(Canvas, ScreenLocation, EntityId, Actor->GetName(), true);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:641]:
```suggestion
				DrawTag(Canvas, ScreenLocation, EntityId, Actor->GetName(), true /*bCentre*/ );
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp [hunk 4]
```diff
@@ -643,7 +675,7 @@ void ASpatialDebugger::DrawDebugLocalPlayer(UCanvas* Canvas)
 	for (int32 i = 0; i < ActorsToDisplay.Num(); ++i)
 	{
 		const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(ActorsToDisplay[i]);
-		DrawTag(Canvas, ScreenLocation, EntityId, ActorsToDisplay[i]->GetName());
+		DrawTag(Canvas, ScreenLocation, EntityId, ActorsToDisplay[i]->GetName(), false);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:678]:
```suggestion
		DrawTag(Canvas, ScreenLocation, EntityId, ActorsToDisplay[i]->GetName(), false /*bCentre*/ );
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp (hunk 2 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:483]: The Authority and the AuthorityIntent have an icon and text so they need two widths. The icons are a fixed size and the text depends on the number of digits in the ID.

Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:483]: I will add some comments to make it clearer.

Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:485]: I've added comments and updated variable names to make the tag width calculation clearer.

**Review by m-samiec — APPROVED** [review]

**Pull request merged**